src/mimefactory.rs

```diff
@@ -2369,7 +2371,16 @@ mod tests {
                 .count(),
             1
         );
-        assert_eq!(part.match_indices("From:").count(), 1);
+
+        // It would be nicer if DC included the From: header here in order to also sign it.
+        // However, making sure that the From: header is included here,
+        // but not duplicated in unencrypted and unsigned messages
+        // and without leaking a cleartext DisplayName in securejoin messages
+        // was deemed not worth the effort and additional code complexity.
+        // (Context: In verified chats, the DisplayName should not be leaked,
+        // so we must make sure not to include it in securejoin messages)
+        assert_eq!(part.match_indices("From:").count(), 0);
```

Comment:
With another half an hour or so, it would surely be possible to find a way not to have to change this test, but I don't think we should introduce more code complexity in this critical area without any real need.